xapian-bindings/java/docs/examples/SimpleSearch.java

```diff
@@ -52,15 +52,12 @@ public static void main(String[] args) throws Exception {
         Enquire enquire = new Enquire(db);
         enquire.setQuery(query);
         MSet matches = enquire.getMSet(0, 2500);    // get up to 2500 matching documents
-        MSetIterator itr = matches.begin();
 
         System.err.println("Found " + matches.size() + " matching documents using " + query);
-        while (itr.hasNext()) {
-            int percent = itr.getPercent();
-            long docID = itr.next();
-            // TODO:  Make this more like a Java Iterator by returning some
-            // kind of "MatchDescriptor" object
-            Document doc = db.getDocument(docID);
+        for (MSetIterator.MatchDescriptor match : matches) {
+            int percent = match.getPercent();
+            long docID = match.getDocId();
+            Document doc = match.getDocument();
             System.err.println(percent + "% [" + docID + "] " + doc.getValue(0));
```

Comment:
I'd be inclined to not even bother using local variables, given each is used only once.